src/app/loadout-builder/processWorker/processUtils.test.ts

```diff
@@ -0,0 +1,220 @@
+import { canTakeAllSeasonalMods, sortProcessModMetadataOrProcessItem } from './processUtils';
+import { ProcessModMetadata, ProcessItem } from './types';
+import { DestinyEnergyType } from 'bungie-api-ts/destiny2';
+
+function getMod(season: number, tag: string, energyType: DestinyEnergyType): ProcessModMetadata {
+  return { season, tag, energyType };
+}
+
+function getItem(
+  season: number,
+  energyType: DestinyEnergyType,
+  compatibleModSeasons?: string[]
+): ProcessItem {
+  return {
+    bucketHash: 0,
+    id: 'id',
+    type: 'type',
+    name: 'name',
+    sockets: null,
+    energyType,
+    basePower: 1050,
+    stats: {},
+    season,
+    compatibleModSeasons,
+  };
+}
+
+/* Mods need to be sorted by sortSeasonalModsOrItems. Items are sorted inside canTakeAllSeasonalMods */
+
+/*
+  Can take all the mods cases
+*/
+describe('Can slot seasonal mods', () => {
+  it('passes for correct seasons with correct energy', () => {
+    const mods = [
+      getMod(11, 'arrivals', 3),
+      getMod(11, 'arrivals', 3),
+      getMod(10, 'worthy', 3),
+    ].sort(sortProcessModMetadataOrProcessItem);
+
+    const items = [
+      getItem(11, 3, ['arrivals', 'worthy']),
+      getItem(11, 3, ['arrivals', 'worthy']),
+      getItem(9, 3, ['worthy', 'dawn', 'undying']),
+    ];
+
+    const result = canTakeAllSeasonalMods(mods, items);
```

Comment:
From our discussion today I could scope canTakeAllSeasonalMods down to use `{season: number; energyType: DestinyEnergyType ;compatibleModSeasons?: string[] }` instead of the ProcessItem so that we don't mock it?

Could also just delete it if you want, I largely used it to ensure what I was doing was correct.